extensions/rich_text_components/Video/directives/oppia-noninteractive-video.component.ts

```diff
@@ -0,0 +1,121 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for the Video rich-text component.
+ *
+ * IMPORTANT NOTE: The naming convention for customization args that are passed
+ * into the directive is: the name of the parameter, followed by 'With',
+ * followed by the name of the arg.
+ */
+
+import { Component, ElementRef, Input, OnInit } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { AutoplayedVideosService } from 'services/autoplayed-videos.service';
+import { ContextService } from 'services/context.service';
+import { HtmlEscaperService } from 'services/html-escaper.service';
+import { ServicesConstants } from 'services/services.constants';
+
+let apiLoaded = false;
```

Comment:
Mistake?

Reply:
Or why this needs to be outside the component?

Reply:
This only changes once during the lifetime of application. If it is inside the component we would end up fetching and loading the youtube embed script every-time time this component was created. If this were a feature used in multiple places, keeping it in a service would be a better place. But currently  restricted to this component. This syntax/ code is taken from the example given in the angular repo. https://github.com/angular/components/tree/master/src/youtube-player#example

Reply:
Okay, thanks.

Reply:
Added